go-selinux/selinux_linux.go

```diff
@@ -1181,6 +1183,101 @@ func dupSecOpt(src string) ([]string, error) {
 	return dup, nil
 }
 
+// checkGroup returns true if group's GID is in the list of GIDs gids.
+func checkGroup(group string, gids []string, lookupGroup func(string) (*user.Group, error)) bool {
+	grp, err := lookupGroup(group)
+	if err != nil {
+		return false
+	}
+
+	for _, gid := range gids {
+		if grp.Gid == gid {
+			return true
+		}
+	}
+	return false
+}
+
+func getSeUserFromReader(username string, gids []string, r io.Reader, lookupGroup func(string) (*user.Group, error)) (string, string, error) {
+	var defaultSeUser, defaultLevel string
+	var groupSeUser, groupLevel string
+
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		line = strings.TrimSpace(line)
+		if line == "" || strings.HasPrefix(line, "#") {
+			continue
+		}
+		parts := strings.SplitN(line, ":", 3)
+		if len(parts) < 2 {
+			continue
+		}
+		userField := parts[0]
+		if userField == "" {
+			continue
+		}
+		seUserField := parts[1]
+		if seUserField == "" {
+			continue
+		}
```

Comment:
I'm not familiar with the spec - are lines matching these conditions valid? If not, it would probably be better to error.

Reply:
Added errors and tests